src/table/data-grid-interaction-model.class.ts

```diff
@@ -0,0 +1,263 @@
+		switch (event.key) {
+			case "Right": // IE specific value
+			case "ArrowRight":
+				event.preventDefault();
+				// add the colspan since getColumnFromCell will return the
+				// first column containing the cell (of N columns it may span)
+				// and we want to navigate to the next "real" column
+				this.goToColumn(this.tableAdapter.findColumnIndex(currentCell) + currentCell.colSpan);
+				break;
+			case "Left": // IE specific value
+			case "ArrowLeft":
+				event.preventDefault();
+				// we only ever need to subtract 1 from the column, since getColumnFromCell returns the
+				// first of N columns containing the cell
+				this.goToColumn(this.tableAdapter.findColumnIndex(currentCell) - 1);
+				break;
+			case "Down": // IE specific value
+			case "ArrowDown":
+				event.preventDefault();
+				this.goToRow(this.currentRow + 1);
+				break;
+			case "Up": // IE specific value
+			case "ArrowUp":
+				event.preventDefault();
+				this.goToRow(this.currentRow - 1);
+				break;
+			case "Home":
+				event.preventDefault();
+				if (event.ctrlKey) {
+					this.goTo({row: 0, column: 0});
+				} else {
+					this.goToColumn(0);
+				}
+				break;
+			case "End":
+				event.preventDefault();
+				if (event.ctrlKey) {
+					this.goTo({ row: this.lastRow, column: this.lastColumn });
+				} else {
+					this.goToColumn(this.lastColumn);
+				}
+				break;
+		}
+	}
+
+	/**
+	 * Handles moving the position to the clicked cell
+	 *
+	 * @param event the MouseEvent to handle
+	 */
+	handleClickEvent(event: MouseEvent) {
+		const cell = (event.target as HTMLElement).closest("td, th") as HTMLTableCellElement;
+		const [rowIndex, cellIndex] = this.tableAdapter.findIndex(cell);
+		this.goTo({ row: rowIndex, column: cellIndex });
+	}
+
+	/**
+	 * Jump to a specific column without changing the row
+	 *
+	 * @param index column to jump to
+	 */
+	goToColumn(index: number) {
+		if (index > this.lastColumn || index < 0) { return; }
+		this.goTo({ row: this.currentRow, column: index});
+	}
+
+	/**
+	 * Jump to a specific row without chaning the column
+	 *
+	 * @param index row to jump to
+	 */
+	goToRow(index: number) {
+		if (index > this.lastRow || index < 0) { return; }
+		this.goTo({row: index, column: this.currentColumn});
+	}
+
+	/**
+	 * Jump to the specified row and column
+	 *
+	 * @param param0 an object that contains `row` and `column` properties
+	 */
+	goTo({row, column}) {
+		this.rowSubject.next({ current: row, previous: this.currentRow });
+		this.columnSubject.next({ current: column, previous: this.currentColumn });
+	}
+
+	/**
+	 * Convenience method to reset the tab indexes on a standard carbon table.
+	 * For custom tables you may want to reset the indexes manually and simply call `.reset()`
+	 */
+	resetTabIndexes() {
+		for (let i = 0; i < this.tableAdapter.lastRow; i++) {
+			const row = this.tableAdapter.getRow(i) as HTMLTableRowElement;
+			for (const cell of Array.from(row.cells)) {
```

Comment:
```suggestion
			for (const cell of row.cells) {
```

should also work.

Reply:
Same as above ... will with the change to return `TableRowAdapter[]` from `getRow`